wit/guest.wit

```diff
@@ -1,10 +1,10 @@
-interface messaging-guest {
-    use messaging-types.{message, guest-configuration, error};
+interface incoming-handler {
+    use types.{message, error};
 
-    /// Returns the list of channels (and extension metadata within guest-configuration) that 
-    /// this component should subscribe to and be handled by the subsequent handler within guest-configuration
-    configure: func() -> result<guest-configuration, error>;
-
-    /// Whenever this guest receives a message in one of the subscribed channels, the message is sent to this handler
-    handler: func(ms: list<message>) -> result<_, error>;
+    /// Whenever this guest receives a message in one of the subscribed channels, the message is
+    /// sent to this handler. The guest is responsible for matching on the channel and handling the
+    /// message accordingly. An Ok result indicates that the message was handled successfully and
+    /// should be considered processed. If explicitly abandoning the message, the guest should return
+    /// error::abandoned with a message indicating why.
+    handle: func(ms: message) -> result<_, error>;
```

Comment:
I talked about this in my reply to Luke in another comment, but I believe the `handle` function should maybe no longer have a return type. Instead, implementors (i.e., guests) should either: (1) do nothing with a received message, or (2) explicitly `reply`. 

Otherwise, I think we can't get away from implicit replies - right? 'Cause who else (other than a host) could be looking at the return type of `handle`?

However, note that, with `request-reply` being an optional interface, the `handle`r may have to instead use `send` over `reply` if `request-reply` is unimplemented.